pkg/kapp/diff/versioned_resource.go

```diff
@@ -91,6 +91,34 @@ func (d VersionedResource) updateAffected(rule ctlconf.TemplateRule, rs []ctlres
 		}
 	}
 
+	for _, res := range rs {
+		for k, v := range res.Annotations() {
+			if k == explicitReferenceKey || strings.HasPrefix(k, explicitReferenceKeyPrefix) {
+				explicitRef := NewExplicitVersionedRef(k, v)
+				objectRef, err := explicitRef.AsObjectRef()
+				if err != nil {
+					return err
+				}
+
+				// Passing empty TemplateAffectedObjRef as explicit references do not have a special name key
+				err = d.buildObjRefReplacementFunc(ctlconf.TemplateAffectedObjRef{})(objectRef)
+				if err != nil {
+					return fmt.Errorf("Error updating explicit versioned reference: %s", err)
```

Comment:
`Error updating explicit versioned reference: %s` => `Updating versioned explicit reference: %s`

Reply:
Oops missed this. Made this change

Reply:
Making this something like `"Parsing versioned explicit ref on resource` as `buildObjRefReplacementFunc` only returns unmarshalling and marshalling errors. Even though it's unlikely to error out after getting parsed in line 100